.travis.yml

```diff
@@ -17,21 +17,22 @@ matrix:
       env:
         - CS_CHECK=true
     - php: 7.1
-    - php: hhvm 
+    - php: hhvm
   allow_failures:
     - php: hhvm
 
 before_install:
   - if [[ $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini || return 0 ; fi
   - composer self-update
   - if [[ $TEST_COVERAGE == 'true' ]]; then composer require --dev --no-update satooshi/php-coveralls:^1.0 ; fi
+  - if [[ $TEST_COVERAGE == 'true' ]]; then composer require --dev --no-update michaelmoussa/php-coverage-checker:^1.1.0 ; fi
 
 install:
   - travis_retry composer install --no-interaction
   - composer info -i
 
 script:
-  - if [[ $TEST_COVERAGE == 'true' ]]; then composer test-coverage ; else composer test ; fi
+  - if [[ $TEST_COVERAGE == 'true' ]]; then composer test-coverage && vendor/bin/php-coverage-checker clover.xml 100 ; else composer test ; fi
```

Comment:
What you present above is basically what we do with coveralls, too:

```yaml
- if [[ $TEST_COVERAGE != 'true' ]]; then composer test ; fi
- if [[ $TEST_COVERAGE == 'true' ]]; then composer test-coverage ; fi
- if [[ $TEST_COVERAGE == 'true' ]]; then composer check-coverage ; fi
```

The main difference currently is that we usually have an `upload-coverage` script we invoke during the `after_script` hook, which essentially uploads to coveralls, so that it can do its own status update on the PR. With your approach, we'd have it in the `script` hook, as it becomes the last check required to pass the job.

Your approach to adding the library during `before_install` is also how we add the coveralls support, so I'm fully on-board that; no reason to add it as a formal dev dependency.